tests/ui/from_instead_of_into.rs

```diff
@@ -0,0 +1,16 @@
+// run-rustfix
+#![warn(clippy::from_instead_of_into)]
+use std::convert::TryFrom;
+
+fn foo<T>(a: T)
+where
+    u32: From<T>,
```

Comment:
Please also add tests where `From` is not the only predicate in the bound. (Also put it in different positions)